autocertkit/status.py

```diff
@@ -90,18 +91,18 @@ def main():
         print "5:An error has occured reading. %s" % TEST_FILE
         sys.exit(1)
 
-    p, f, s, w = ack_run.get_status()
+    p, f, s, w, r = ack_run.get_status()
 
-    if w == 0:
+    if w+r == 0:
         print "0:Finished (Passed:%d, Failed:%d, Skipped:%d)" % (p, f, s)
-    elif not running and utils.get_reboot_flag():
-        print "3:Server rebooting... (Passed:%d, Failed:%d, Skipped:%d, Waiting:%d)" % (p, f, s, w)
-    elif not running and not utils.get_reboot_flag():
-        print "1:Process not running. An error has occurred. (Passed:%d, Failed:%d, Skipped: %d, Waiting:%d)" % (p, f, s, w)
+    elif not running and uptime_seconds <= 600 and r > 0:
```

Comment:
Sorry I am not aware of my reply disappeared, but I think you should already received github notification mail.
If not please let me know, I will re-add.

Reply:
See my [section 3 of design](https://info.citrite.net/display/xenserver/REQ-46%3A+HCL+design+for+SR-IOV+pass-through+support#REQ-46:HCLdesignforSR-IOVpass-throughsupport-2.10ParseandshowSR-IOVinformation)
In my mind test should have no reboot status, but just 'init' 'running' and 'done', ACK is similar, it will be more easier to understand if ACK run on another host, not in pool, like xenrt controller, I don't think ACK should 'reboot' during once test run.

You can see from init.d script, ACK would not stop.
```
# cat /etc/rc3.d/S99auto-cert-kit
...
stop() {
#Nothing, do nothing
log "Stopping"
}
```
So once user runs ack_cli.py to statup ACK, ACK should be always in running status, until done.
'reboot' is just one step in test case itself, it requires host reboot, but not to ACK.

There is time gap between OS and ACK when start and stop indeed, and we can use reasonable threshold to handle.
In my mind 5 to 10 minutes is acceptable at present, because ACK would not run in production environment.
Oppositely if ACK still use/check reboot flag file and vendor's certain driver caused XenServer start/stop too slow, then ACK could not be aware of. That's not good.

Reply:
> In my mind test should have no reboot status, but just 'init' 'running' and 'done', ACK is similar, it will be more easier to understand if ACK run on another host, not in pool, like xenrt controller, I don't think ACK should 'reboot' during once test run.

OK - so this is really to pave the way for having the ACK runner permanently active, whilst having remote hosts able to reboot as required?

Reply:
At present ACK does not support to run on independent server yet, I just explain the reason that ACK rebooting is needless to expose to user.